File: c++代码/530.二叉搜索树的最小绝对差.cpp
#include <iostream>
#include <vector>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};
class Solution
{
private:
    vector<int> num_lst;

public:
    int getMinimumDifference(TreeNode *root)
    {
        zhongxu(root);
        int min_num = INT32_MAX;
        for (int i = 0; i < num_lst.size()-1; i++)
        {
            min_num = min_num < abs(num_lst[i] - num_lst[i+1]) ? min_num : abs(num_lst[i] - num_lst[i+1]);
        }
        return min_num;
        
    }
    void zhongxu(TreeNode* root)
    {
        if (root->left) zhongxu(root->left);
        if (root) this->num_lst.push_back(root->val);
        if (root->right) zhongxu(root->right);
    }
};

File: 力扣/每日一题/530. 二叉搜索树的最小绝对差.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def getMinimumDifference(self, root: TreeNode) -> int:
        num_lst = []
        def zhongxu(root:TreeNode):
            if root.left:
                zhongxu(root.left)
            num_lst.append(root.val)
            if root.right:
                zhongxu(root.right)

        zhongxu(root)
        min_num = float('INF')
        for i in range(len(num_lst) -1):
            min_num = min_num if min_num < abs(num_lst[i] - num_lst[i+1]) else abs(num_lst[i] - num_lst[i+1])
        return min_num